### PR TITLE
bump gradle plugin version

### PR DESCRIPTION
**Pull request by Zayankovsky**
### Description
Refs #5102. 
Bump Gradle plugin version. 
I'll try to update as few libraries as possible and split the updates into several PRs.

### CODE REVIEW

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesControllerTest.kt [hunk 1]
```diff
@@ -403,17 +403,17 @@ class RouteAlternativesControllerTest {
             )
         )
 
-        val routeProgressSlot = slot<RouteProgress>()
-        val alternativesSlot = slot<List<DirectionsRoute>>()
-        val routerOriginSlot = slot<RouterOrigin>()
+        val routeProgressSlots = mutableListOf<RouteProgress>()
```
Comment (RingerJK) [inline libnavigation-core/src/test/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesControllerTest.kt:406]: why need to replace slot by list if need the latest item only? 🤔

Comment (Zayankovsky) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesControllerTest.kt:406]: It is a new limitation in mockk library: https://github.com/mockk/mockk/releases/tag/1.10.3:

> issue [#352](https://github.com/mockk/mockk/issues/352) prevent slots from being used when verifying multiple calls

File: gradle.properties [hunk 2]
```diff
@@ -23,6 +23,7 @@ POM_SNAPSHOT_VERSION_NAME=2.1.0-SNAPSHOT
 # Specifies the JVM arguments used for the daemon process.
 # The setting is particularly useful for tweaking memory settings.
 org.gradle.jvmargs=-Xmx2048M
+systemProp.kotlin.daemon.jvm.options=-Xss8m
```
Comment (LukasPaczos) [inline gradle.properties:26]: Could you add a comment here explaining or linking to why we need to increase the stack size?

Just so we know where this comes from down the line.

**Review by LukasPaczos — APPROVED** [review]
:ship:

**Review by RingerJK — APPROVED** [review]

**Pull request merged**